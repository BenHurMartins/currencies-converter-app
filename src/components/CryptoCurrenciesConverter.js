import { useState } from "react";
import '../styles.css';
import Selector from "./Selector";
import { convertCurrencyCrypto } from "../api";
import { i18n } from "../i18n";

const CryptoCurrenciesConverter = (props) => {

    const [fromCurrencyCrypto, setFromCurrencyCrypto] = useState('BTC')
    const [toCurrencyCrypto, setToCurrencyCrypto] = useState('USD')
    const [fromValueCrypto, setFromValueCrypto] = useState(0)
    const [toValueCrypto, setToValueCrypto] = useState(0)

    const handleFromValueCryptoChange = (e) => {
        const value = e.target.value.replace(',', '.')
        if (!isNaN(value)) {
            setFromValueCrypto(value)
        }
    }

    const handleFromCurrencyChangeCrypto = (e) => {
        setFromValueCrypto(0)
        setFromCurrencyCrypto(e.target.value);
    }
    const handleToCurrencyChangeCrypto = (e) => {
        setToValueCrypto(0)
        setToCurrencyCrypto(e.target.value);
    }


    const handleCryptoConversion = () => {
        const value = parseFloat(fromValueCrypto).toFixed(2) == 0 ? parseFloat(1).toFixed(2) : parseFloat(fromValueCrypto).toFixed(2)
        setFromValueCrypto(value)
        convertCurrencyCrypto(fromCurrencyCrypto, toCurrencyCrypto, value)
            .then(result => {
                setToValueCrypto(result)
            })
    }

    return (
        <>
            <p className="app-subtitle">
                {i18n.cryptoCurrencySubtitle[props.lang]}
            </p>
            <div className="main-container">
                <div className="form-container">
                    <div className="half-container">
                        <p className="label">{i18n.from[props.lang]}</p>
                        <input placeholder="0.00" min="0" value={fromValueCrypto} onChange={handleFromValueCryptoChange} />
                        <Selector value={fromCurrencyCrypto} onChange={handleFromCurrencyChangeCrypto} options={props.cryptoCurrenciesArray} currencies={props.cryptos} />
                    </div>
                    <div className="half-container">
                        <p className="label">{i18n.to[props.lang]}</p>
                        <input placeholder="0.00" value={toValueCrypto} />
                        <Selector value={toCurrencyCrypto} onChange={handleToCurrencyChangeCrypto} options={props.currenciesArray} currencies={props.currencies} />
                    </div>
                </div>
                <div>
                    <button onClick={handleCryptoConversion}>{i18n.cryptoCurrencyConvertButton[props.lang]}</button>
                </div>
            </div>
        </>
    )
}

export default CryptoCurrenciesConverter;