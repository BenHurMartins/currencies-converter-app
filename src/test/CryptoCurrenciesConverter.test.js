import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CryptoCurrenciesConverter from "../components/CryptoCurrenciesConverter";
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


describe('CryptoCurrencyConverter', () => {

    
    it('should render the component', ()=> {
        const optionsArray=[
            'USD', 'BRL'
        ]
        const optionsCryptoArray=[
            'BTC', 'ETH'
        ]
        const currencies = {
            'USD': 'United States Dolar',
            'BRL': 'Brazilian Real'
        }
        const cryptos = {
            'BTC': 'Bitcoin',
            'ETH': 'Ethereum'
        }
        const component = renderer.create(<CryptoCurrenciesConverter lang={'EN'} cryptoCurrenciesArray={optionsCryptoArray} currencies={currencies} currenciesArray={optionsArray} cryptos={cryptos}/>)
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should show the text in english', () => {
        const cryptoInstance = shallow(<CryptoCurrenciesConverter lang={'EN'} />);
        const element = cryptoInstance.find('.app-subtitle');
        expect(element.text()).toBe('Crypto Currency');
    }); 
    
    it('should be in portuguese', () => {
        const cryptoInstance = shallow(<CryptoCurrenciesConverter lang={'PT_BR'} />);
        const element = cryptoInstance.find('.app-subtitle');
        expect(element.text()).toBe('Cripto Moeda');
    })
})