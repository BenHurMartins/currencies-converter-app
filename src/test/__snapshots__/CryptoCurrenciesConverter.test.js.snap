// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoCurrencyConverter should render the component 1`] = `
Array [
  <p
    className="app-subtitle"
  >
    Crypto Currency
  </p>,
  <div
    className="main-container"
  >
    <div
      className="form-container"
    >
      <div
        className="half-container"
      >
        <p
          className="label"
        >
          From:
        </p>
        <input
          className="input-from"
          min="0"
          onChange={[Function]}
          placeholder="0.00"
          value={0}
        />
        <select
          onChange={[Function]}
          value="BTC"
        >
          <option
            value="BTC"
          >
            Bitcoin
          </option>
          <option
            value="ETH"
          >
            Ethereum
          </option>
        </select>
      </div>
      <div
        className="half-container"
      >
        <p
          className="label"
        >
          To:
        </p>
        <input
          className="input-to"
          placeholder="0.00"
          value={0}
        />
        <select
          onChange={[Function]}
          value="USD"
        >
          <option
            value="USD"
          >
            United States Dolar
          </option>
          <option
            value="BRL"
          >
            Brazilian Real
          </option>
        </select>
      </div>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        Convert Crypto
      </button>
    </div>
  </div>,
]
`;
